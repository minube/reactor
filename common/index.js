import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
