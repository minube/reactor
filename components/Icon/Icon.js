import {
  bool, number, oneOfType, string,
} from 'prop-types';
import React from 'react';
import { Platform, StyleSheet, Image } from 'react-native';

import ASSETS from './assets';
import styles from './Icon.style';

const isWeb = Platform.OS === 'web';

const Icon = ({
  color, invert, size, value, ...inherit
}) => {
  const [, navigation] = value.split('nav');
  let asset;
  let rotate;

  if (navigation) {
    asset = ASSETS.navArrow;
    if (navigation === 'Down') rotate = 90;
    else if (navigation === 'Left') rotate = -180;
    else if (navigation === 'Up') rotate = 270;
  } else {
    asset = ASSETS[value];
    if (!asset) asset = typeof value === 'string' ? { uri: value } : value;
  }

  return (
    <Image
      resizeMode="contain"
      source={asset}
      style={[
        styles.container,
        inherit.style,

        (color || invert || rotate || size) && StyleSheet.flatten([
          color && !isWeb && { tintColor: color },
          invert && isWeb && { filter: 'invert(100%)' },
          rotate && { transform: [{ rotate: `${rotate}deg` }] },
          size && { width: size, height: size },
        ]),
      ]}
    />
  );
};

Icon.propTypes = {
  color: string,
  invert: bool,
  size: number,
  value: oneOfType([number, string]),
};

Icon.defaultProps = {
  color: undefined,
  invert: false,
  size: undefined,
  value: 'errorOutline',
};

export default Icon;
