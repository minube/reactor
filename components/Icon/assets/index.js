import close from './close.png';
import closeContrast from './close_contrast.png';
import doneContrast from './done_contrast.png';
import email from './email.png';
import error from './error.png';
import errorOutline from './error_outline.png';
import facebook from './facebook.png';
import google from './google.png';
import left from './left.png';
import linkedin from './linkedin.png';
import navArrow from './nav-arrow.png';
import right from './right.png';
import search from './search.png';
import shareContrast from './share_contrast.png';
import twitter from './twitter.png';

export default {
  close,
  closeContrast,
  doneContrast,
  email,
  error,
  errorOutline,
  facebook,
  google,
  left,
  linkedin,
  navArrow,
  right,
  search,
  shareContrast,
  twitter,
};
