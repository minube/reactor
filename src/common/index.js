import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
