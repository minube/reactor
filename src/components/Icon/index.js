import { array, number, oneOfType, string } from 'prop-types';
import React from 'react';
import { Platform, Image } from 'react-native';

import ASSETS from './assets';
import styles from './style';

const Icon = ({ style, value }) => (
  <Image
    resizeMode="contain"
    source={
      Platform.select({
        web: {
          uri: ASSETS[value],
        },
      })
    }
    style={[styles.icon, style]}
  />
);

Icon.propTypes = {
  style: oneOfType([array, number]),
  value: string,
};

Icon.defaultProps = {
  style: [],
  value: undefined,
};

export default Icon;
