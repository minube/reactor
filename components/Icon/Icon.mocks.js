import { THEME } from '../../common';

const COLOR = THEME.COLOR.PRIMARY;

const ICONS = [
  'close',
  'closeContrast',
  'doneContrast',
  'email',
  'error',
  'errorOutline',
  'expand',
  'facebook',
  'google',
  'left',
  'leftOutline',
  'linkedin',
  'right',
  'rightOutline',
  'shareContrast',
  'twitter',
];

const STYLE = {
  backgroundColor: THEME.COLOR.PRIMARY,
};

const VALUE = ICONS[0];
const VALUE_URI = 'http://soyjavi.com/assets/images/soyjavi.jpg';

export default {
  COLOR, ICONS, STYLE, VALUE, VALUE_URI,
};
