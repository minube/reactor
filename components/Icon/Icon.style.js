import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const { COLOR, FONT } = THEME;

export default StyleSheet.create({
  container: {
    backgroundColor: COLOR.TRANSPARENT,
    height: FONT.SIZE.LARGE,
    tintColor: COLOR.ALERT,
    width: FONT.SIZE.LARGE,
  },
});
