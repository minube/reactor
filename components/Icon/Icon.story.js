import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { boolean, color, number, select } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import Icon from './Icon';

storiesOf('✅ Icon', module)
  .add('default', () => (
    <Icon />
  ))
  .add('value', () => (
    <Icon value="closeDark" />
  ))
  .add('value (as uri)', () => (
    <Icon value="https://api.adorable.io/avatars/48/8" />
  ))
  .add('color (only Native)', () => (
    <withNotes notes="Color is only supported in Native environments">
      <Icon color={THEME.COLOR.ERROR} value="closeDark" />
    </withNotes>
  ))
  .add('invert (only Web)', () => (
    <withNotes notes="invert is only supported in web environments">
      <Icon invert value="closeDark" />
    </withNotes>
  ))
  .add('size', () => (
    <Icon size={48} value="closeDark" />
  ))
  .add('style', () => (
    <Icon style={{ backgroundColor: THEME.COLOR.SECONDARY }} value="closeDark" />
  ))
  .add('🏀 Playground', () => (
    <Icon
      size={number('size', 16)}
      color={color('color', THEME.COLOR.TEXT)}
      invert={boolean('invert', false)}
      value={select('value', ['apps', 'close', 'closeDark', 'emailDark', 'left', 'right', 'share'], 'closeDark')}
    />
  ));

