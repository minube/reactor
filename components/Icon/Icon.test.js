import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Icon from './Icon';

const dataSource = [
  { caption: 'Spain', href: '/' },
  { caption: 'Basque Country', href: '/' },
  { caption: 'Bilbao', href: '/' },
];

const ICON_VALUE = 'close';

describe('<Icon>', () => {
  it('renders', () => {
    const tree = renderer.create(<Icon value={ICON_VALUE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value} as uri', () => {
    const tree = renderer.create(<Icon value="https://api.adorable.io/avatars/48/8" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color} (only Native)', () => {
    const tree = renderer.create(<Icon value={ICON_VALUE} color={THEME.COLOR.ERROR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {invert} (only Web)', () => {
    const tree = renderer.create(<Icon value={ICON_VALUE} invert />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {size}', () => {
    const tree = renderer.create(<Icon value={ICON_VALUE} size={48} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Icon value={ICON_VALUE} style={{ backgroundColor: THEME.COLOR.ACCENT }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

