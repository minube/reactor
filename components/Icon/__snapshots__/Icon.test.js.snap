// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> renders 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/error_outline.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Icon> when {color} (only Native) 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/close.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      Object {
        "tintColor": "#10c0bc",
      },
    ]
  }
/>
`;

exports[`<Icon> when {invert} (only Web) 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/close.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      Object {},
    ]
  }
/>
`;

exports[`<Icon> when {size} 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/close.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      Object {
        "height": 48,
        "width": 48,
      },
    ]
  }
/>
`;

exports[`<Icon> when {style} 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/close.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      Object {
        "backgroundColor": "#10c0bc",
      },
      undefined,
    ]
  }
/>
`;

exports[`<Icon> when {value} (as uri) 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "uri": "https://api.adorable.io/avatars/48/8",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Icon> when {value} (navDown) 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/nav-arrow.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      Object {
        "transform": Array [
          Object {
            "rotate": "90deg",
          },
        ],
      },
    ]
  }
/>
`;

exports[`<Icon> when {value} (navLeft) 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/nav-arrow.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      Object {
        "transform": Array [
          Object {
            "rotate": "-180deg",
          },
        ],
      },
    ]
  }
/>
`;

exports[`<Icon> when {value} (navRight) 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/nav-arrow.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Icon> when {value} (navUp) 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/nav-arrow.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      Object {
        "transform": Array [
          Object {
            "rotate": "270deg",
          },
        ],
      },
    ]
  }
/>
`;

exports[`<Icon> when {value} 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../components/Icon/assets/close.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      },
      undefined,
      undefined,
    ]
  }
/>
`;
